src/routes/choice-exercises/components/SingleChoiceExercise.tsx

```diff
@@ -107,36 +107,63 @@ const ChoiceExerciseScreen = ({
       return
     }
     setSelectedAnswer(clickedAnswer)
+
+    const { nthRetry } = getRetryInfoOfCurrent()
+
     const correctSelected = [correctAnswer, ...currentDocument.alternatives].find(it =>
       isAnswerEqual(it, clickedAnswer)
     )
 
     if (correctSelected !== undefined) {
       setCorrectAnswer(clickedAnswer)
-      const result: DocumentResultType = { ...currentDocument, result: SIMPLE_RESULTS.correct }
-      setResults([...results, result])
+      const result: DocumentResultType = {
+        ...currentDocument,
+        result: SIMPLE_RESULTS.correct,
+        numberOfTries: nthRetry + 1
+      }
+      updateResult(result)
     } else {
-      const result: DocumentResultType = { ...currentDocument, result: SIMPLE_RESULTS.incorrect }
-      setResults([...results, result])
+      const result: DocumentResultType = {
+        ...currentDocument,
+        result: SIMPLE_RESULTS.incorrect,
+        numberOfTries: nthRetry + 1
+      }
+      updateResult(result)
     }
     setTimeout(() => {
       setDelayPassed(true)
     }, correctAnswerDelay)
   }
 
+  const updateResult = (result: DocumentResultType): void => {
+    const indexOfCurrentResult = results.findIndex(result => result.id === currentDocument?.id)
+    indexOfCurrentResult !== -1 ? (results[indexOfCurrentResult] = result) : setResults([...results, result])
+  }
+
+  const getRetryInfoOfCurrent = (): { nthRetry: number; needsToBeRepeated: boolean } => {
+    const indexOfCurrent = results.findIndex(result => result.id === currentDocument?.id)
+    const nthRetry = indexOfCurrent === -1 ? 0 : results[indexOfCurrent].numberOfTries
+    const needsToBeRepeated =
+      nthRetry < 3 && (indexOfCurrent === -1 || results[indexOfCurrent].result === SIMPLE_RESULTS.incorrect)
+    return { nthRetry, needsToBeRepeated }
```

Comment:
Something should work as well:
```suggestion
   return results.find(result => result.id === currentDocument?.id).map(({ numberOfTries, result }) => ({ numberOfTries, needsToBeRepeated: numberOfTries < 3 && result === SIMPLE_RESULTS.incorrect }))
```